wp-session-manager.php

```diff
@@ -3,33 +3,26 @@
  * Plugin Name: WP Session Manager
  * Plugin URI:  http://jumping-duck.com/wordpress/plugins
  * Description: Prototype session management for WordPress.
- * Version:     1.2.1
+ * Version:     2.0
  * Author:      Eric Mann
  * Author URI:  http://eamann.com
  * License:     GPLv2+
  */
 
-// let users change the session cookie name
-if( ! defined( 'WP_SESSION_COOKIE' ) ) {
-	define( 'WP_SESSION_COOKIE', '_wp_session' );
-}
+require __DIR__ . '/vendor/autoload.php';
 
-if ( ! class_exists( 'Recursive_ArrayAccess' ) ) {
-	include 'includes/class-recursive-arrayaccess.php';
-}
-
-// Include utilities class
-if ( ! class_exists( 'WP_Session_Utils' ) ) {
-	include 'includes/class-wp-session-utils.php';
-}
+// Queue up the session stack
+EAMann\Sessionz\Manager::initialize()
+	->addHandler( new \EAMann\Sessionz\Handlers\OptionsHandler() )
+	->addHandler( new \EAMann\Sessionz\Handlers\MemoryHandler() );
 
 // Include WP_CLI routines early
 if ( defined( 'WP_CLI' ) && WP_CLI ) {
+	include 'includes/class-wp-session-utils.php';
 	include 'includes/wp-cli.php';
 }
 
-// Only include the functionality if it's not pre-defined.
-if ( ! class_exists( 'WP_Session' ) ) {
-	include 'includes/class-wp-session.php';
-	include 'includes/wp-session.php';
-}
+// Start up session management, if we're not in the CLI
+if ( ! defined( 'WP_CLI' ) || false === WP_CLI ) {
```

Comment:
This could be collapsed into an `else` for the previous conditional (where we loaded the WP-CLI commands if we *are* in a WP-CLI context.